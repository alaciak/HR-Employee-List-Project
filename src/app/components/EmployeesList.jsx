import React from 'react';
import EmployeeData from './EmployeeData.jsx';

class EmployeesList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      employeeList: [],
      loading: true
    };
  }

  componentDidMount() {
    fetch('http://localhost:3000/employees').then(r => r.json()).then(data => {
      this.setState({
        employeeList: data,
        loading: false
      });
    }).catch(function(error) {
      console.log(error);
    });
  }

  onRemoveEmployee = employeeId => {
    let employeeList = this.state.employeeList.slice();
    let newEmployeeList = employeeList.filter((el) => el.id !== employeeId);
    const baseUrl = 'http://localhost:3000/employees'
    fetch(baseUrl + '/' + employeeId, { method : 'DELETE' }).then(r => r.json()).then(data => {
        this.setState({
          employeeList: newEmployeeList
        });
      }).catch(function(error) {
        console.log(error);
      });
    }

  render() {
    if (this.state.loading) {
      return null;
    } else {
      const employeeData = this.state.employeeList.map(employee => {
        return <EmployeeData key={ employee.id } id={ employee.id } employee={ employee } onRemoveEmployee={ this.onRemoveEmployee }/>
      });
      return (
        <section className='employee-list'>
          <div className='row'>
            <div className='container'>
              <table className='employee-list_table col-12'>
                <caption>Employee List</caption>
                <thead className='employee-list_table-heading'>
                  <tr>
                    <th>Full Name</th>
                    <th>Role</th>
                    <th>Experience</th>
                    <th></th>
                  </tr>
                </thead>
                <tbody className='employee-list_table-body'>
                  {employeeData}
                </tbody>
              </table>
            </div>
          </div>
        </section>
      )
    }
  }
}

module.exports = EmployeesList;
