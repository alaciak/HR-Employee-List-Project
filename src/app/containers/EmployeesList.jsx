import React from 'react';
import EmployeeData from './EmployeeData.jsx';
import { connect } from 'react-redux';
import { getList } from '../actions/employeeListActions';

export class EmployeesList extends React.Component {

  componentDidMount() {
    this.props.getList();
  }

  render() {
    if (this.props.loading) {
      return null;
    } else {
      const employeeData = this.props.employeeList.map(employee => {
        return <EmployeeData key={ employee.id } id={ employee.id } employee={ employee }/>
      });
      return (
        <section className='employee-list'>
          <div className='row'>
            <div className='container'>
              <table className='employee-list_table col-12'>
                <caption>Employee List</caption>
                <thead className='employee-list_table-heading'>
                  <tr>
                    <th>Full Name</th>
                    <th>Role</th>
                    <th>Experience</th>
                    <th></th>
                  </tr>
                </thead>
                <tbody className='employee-list_table-body'>
                  { employeeData }
                </tbody>
              </table>
            </div>
          </div>
        </section>
      )
    }
  }
}

const mapStateToProps = (state) => {
  return {
    employeeList: state.employeeListReducer.employeeList,
    loading: state.employeeListReducer.loading
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    getList: () => {
      dispatch(getList());
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(EmployeesList);
