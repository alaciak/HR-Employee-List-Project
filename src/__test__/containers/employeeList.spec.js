import React from 'react';
import { EmployeesList } from '../../app/containers/EmployeesList.jsx';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('EmployeesList', () => {

  it('should not render if the data are not fetched from the server', () => {

    const wrapper = shallow(
      <EmployeesList loading={ true }/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();

  });

  it('should render if the data are fetched from the server', () => {

    const testEmployeeList = [{ id: 1, id: 2 }];

    const wrapper = shallow(
      <EmployeesList loading={ false } employeeList= { testEmployeeList }/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();

  });

});
