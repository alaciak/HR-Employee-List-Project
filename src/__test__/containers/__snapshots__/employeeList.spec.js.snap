// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeesList should not render if the data are not fetched from the server 1`] = `null`;

exports[`EmployeesList should render if the data are fetched from the server 1`] = `
<section
  className="employee-list"
>
  <div
    className="row"
  >
    <div
      className="container"
    >
      <table
        className="employee-list_table col-12"
      >
        <caption>
          Employee List
        </caption>
        <thead
          className="employee-list_table-heading"
        >
          <tr>
            <th>
              Full Name
            </th>
            <th>
              Role
            </th>
            <th>
              Experience
            </th>
            <th />
          </tr>
        </thead>
        <tbody
          className="employee-list_table-body"
        >
          <Connect(withRouter(EmployeeData))
            employee={
              Object {
                "id": 2,
              }
            }
            id={2}
            key="2"
          />
        </tbody>
      </table>
    </div>
  </div>
</section>
`;
